src/payment/strategies/googlepay/googlepay-payment-strategy.spec.ts

```diff
@@ -0,0 +1,341 @@
+} from '../../../checkout';
+import { getCheckoutState } from '../../../checkout/checkouts.mock';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { ConfigActionCreator, ConfigRequestSender } from '../../../config';
+import { getConfigState } from '../../../config/configs.mock';
+import { getCustomerState } from '../../../customer/customers.mock';
+import { OrderActionCreator } from '../../../order';
+import {
+    createPaymentClient,
+    createPaymentStrategyRegistry,
+    PaymentActionCreator,
+    PaymentMethodActionCreator,
+    PaymentStrategyActionCreator
+} from '../../../payment';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodRequestSender from '../../payment-method-request-sender';
+import { getGooglePay, getPaymentMethodsState } from '../../payment-methods.mock';
+import { PaymentInitializeOptions } from '../../payment-request-options';
+import PaymentRequestSender from '../../payment-request-sender';
+import { BraintreeScriptLoader, BraintreeSDKCreator } from '../braintree';
+
+import { GooglePaymentData, GooglePayInitializer, TokenizePayload } from './googlepay';
+import GooglePayBraintreeInitializer from './googlepay-braintree-initializer';
+import GooglePayPaymentProcessor from './googlepay-payment-processor';
+import GooglePayPaymentStrategy from './googlepay-payment-strategy';
+import GooglePayScriptLoader from './googlepay-script-loader';
+import {
+    getGoogleOrderRequestBody,
+    getGooglePaymentDataPayload,
+} from './googlepay.mock';
+
+describe('GooglePayPaymentStrategy', () => {
+    let store: CheckoutStore;
+    let googlePayScriptLoader: GooglePayScriptLoader;
+    let strategy: GooglePayPaymentStrategy;
+    let checkoutActionCreator: CheckoutActionCreator;
+    let paymentMethodActionCreator: PaymentMethodActionCreator;
+    let paymentStrategyActionCreator: PaymentStrategyActionCreator;
+    let paymentActionCreator: PaymentActionCreator;
+    let orderActionCreator: OrderActionCreator;
+    let googlePayInitializer: GooglePayInitializer;
+    let requestSender: RequestSender;
+    let googlePayPaymentProcessor: GooglePayPaymentProcessor;
+    let container: HTMLDivElement;
+    let walletButton: HTMLAnchorElement;
+    let paymentMethodMock: PaymentMethod;
+
+    beforeEach(() => {
+        store = createCheckoutStore({
+            checkout: getCheckoutState(),
+            customer: getCustomerState(),
+            config: getConfigState(),
+            cart: getCartState(),
+            paymentMethods: getPaymentMethodsState(),
+        });
+
+        const checkoutRequestSender = new CheckoutRequestSender(createRequestSender());
+        const configRequestSender = new ConfigRequestSender(createRequestSender());
+        const configActionCreator = new ConfigActionCreator(configRequestSender);
+        const _requestSender: PaymentMethodRequestSender = new PaymentMethodRequestSender(requestSender);
```

Comment:
please remove the underscore (due naming conventions this is for private instance members)